### PR TITLE
v1.6.0 rc1

### PR DESCRIPTION
**Pull request by OleksiiBulba**
- Plugin refactor and refresh.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -1,21 +1,60 @@
...
             "email": "stanislau_komar@epam.com"
+        },
+        {
+            "name": "Oleksii Bulba",
+            "email": "oleksii_bulba@epam.com"
         }
     ],
     "require": {
-        "php": ">=8.0"
+        "php": "^8.2",
```
Comment (Asisyas) [inline composer.json]: Redunant requirement.

File: src/Business/Provider/FactoryProvider.php [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    {
+    }
+
+    public function getFactory(): SerializerFactoryInterface
+    {
+        $iterator = $this->kernel->plugins(SerializerFactoryProviderPluginInterface::class);
+
+        /** @var SerializerFactoryProviderPluginInterface $serializerFactoryProviderPlugin */
+        foreach ($iterator as $serializerFactoryProviderPlugin) {
+            return $serializerFactoryProviderPlugin->provideSerializerFactory();
```
Comment (Asisyas) [inline src/Business/Provider/FactoryProvider.php]: You really wants to return first available plugin ?

File: composer.json (hunk 1 above)
Comment (OleksiiBulba) [reply composer.json]: Removed

File: src/Business/Provider/FactoryProvider.php (hunk 2 above)
Comment (OleksiiBulba) [reply src/Business/Provider/FactoryProvider.php]: MR updated

File: src/Business/Serializer/PrioritizedSerializer.php [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    {
+        foreach ($this->serializers as $serializer) {
+            $serialized = $serializer->serialize($data, $format, $context);
+
+            if (false !== $serialized) {
+                return $serialized;
+            }
+        }
+
+        return false;
```
Comment (Asisyas) [inline src/Business/Serializer/PrioritizedSerializer.php]: throw exception.

File: src/Business/Serializer/PrioritizedSerializer.php [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    {
+        foreach ($this->serializers as $serializer) {
+            $deserialized = $serializer->deserialize($data, $type, $format, $context);
+
+            if (null !== $deserialized) {
+                return $deserialized;
+            }
+        }
+
+        return null;
```
Comment (Asisyas) [inline src/Business/Serializer/PrioritizedSerializer.php]: throw exception

File: src/SerializerPlugin.php [hunk 5]
```diff
@@ -3,51 +3,59 @@
...
+
+    public function getPriority(): int
+    {
+        return -1;
+    }
+
+    public function createSerializer(): SerializerInterface
+    {
+        return new class implements SerializerInterface
+        {
```
Comment (Asisyas) [inline src/SerializerPlugin.php]: The class should be move to a some separate plugin.
For example, if I don't want to use the default serializer in the app at all..

Comment (OleksiiBulba) [reply src/SerializerPlugin.php]: Default serializer removed as unsafe

File: src/Business/Serializer/PrioritizedSerializer.php (hunk 4 above)
Comment (OleksiiBulba) [reply src/Business/Serializer/PrioritizedSerializer.php]: Done

File: src/Business/Serializer/PrioritizedSerializer.php (hunk 3 above)
Comment (OleksiiBulba) [reply src/Business/Serializer/PrioritizedSerializer.php]: Done

File: src/Business/Pool/SerializerPool.php [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+    public function deserialize(mixed $data, SerializerContextInterface $context): mixed
+    {
+        if (\array_key_exists($contextClass = \get_class($context), $this->serializerPoolByContext)) {
+            return $this->serializerPoolByContext[$contextClass]->deserialize($data, $context);
+        }
+
+        $serializers = $this->getSerializers();
+
+        foreach ($serializers as $serializer) {
+            if ($serializer->supports($context)) {
```
Comment (Asisyas) [inline src/Business/Pool/SerializerPool.php]: extra investment.
```
if( ! $serializer->supports($context) ) { continue }
```

File: src/Business/Serializer/SerializerInterface.php [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+namespace Micro\Plugin\Serializer\Business\Serializer;
+
+use Micro\Plugin\Serializer\Business\Context\SerializerContextInterface;
+
+interface SerializerInterface
+{
+    /**
+     * Serializes data in the appropriate format.
+     */
+    public function serialize(mixed $data, SerializerContextInterface $context): string;
```
Comment (Asisyas) [inline src/Business/Serializer/SerializerInterface.php:17]: @throws tag missed.

File: src/Business/Serializer/SerializerInterface.php [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+
+    /**
+     * Deserializes data into the given type.
+     */
+    public function deserialize(mixed $data, SerializerContextInterface $context): mixed;
```
Comment (Asisyas) [inline src/Business/Serializer/SerializerInterface.php:24]: @throws tag missed.

File: src/Business/Serializer/SerializerInterface.php (hunk 7 above)
Comment (OleksiiBulba) [reply src/Business/Serializer/SerializerInterface.php:17]: Added

File: src/Business/Serializer/SerializerInterface.php (hunk 8 above)
Comment (OleksiiBulba) [reply src/Business/Serializer/SerializerInterface.php:24]: Added

File: src/Business/Pool/SerializerPool.php (hunk 6 above)
Comment (OleksiiBulba) [reply src/Business/Pool/SerializerPool.php]: Fixed

**Review by Asisyas — APPROVED** [review]

**Pull request merged**